### ISSUE TITLE
succeed to build torch-sparse,torch-cluster (both latest version I guess) but fail to build torch-scatter

### ISSUE THREAD

**Issue by jzhang538** [rusty1s/pytorch_scatter#150]
Running setup.py bdist_wheel for torch-scatter ... error
  Complete output from command /mnt/lustre/share/spring/envs/r0.2.1/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-rt02l3rf/torch-scatter/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpcvevpyq2pip-wheel- --python-tag cp36:
  No CUDA runtime is found, using CUDA_HOME='/mnt/lustre/share/cuda-10.0'
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.6
  creating build/lib.linux-x86_64-3.6/torch_scatter
  copying torch_scatter/__init__.py -> build/lib.linux-x86_64-3.6/torch_scatter
  copying torch_scatter/placeholder.py -> build/lib.linux-x86_64-3.6/torch_scatter
  copying torch_scatter/scatter.py -> build/lib.linux-x86_64-3.6/torch_scatter
  copying torch_scatter/segment_coo.py -> build/lib.linux-x86_64-3.6/torch_scatter
  copying torch_scatter/segment_csr.py -> build/lib.linux-x86_64-3.6/torch_scatter
  copying torch_scatter/utils.py -> build/lib.linux-x86_64-3.6/torch_scatter
  creating build/lib.linux-x86_64-3.6/torch_scatter/composite
  copying torch_scatter/composite/__init__.py -> build/lib.linux-x86_64-3.6/torch_scatter/composite
  copying torch_scatter/composite/logsumexp.py -> build/lib.linux-x86_64-3.6/torch_scatter/composite
  copying torch_scatter/composite/softmax.py -> build/lib.linux-x86_64-3.6/torch_scatter/composite
  copying torch_scatter/composite/std.py -> build/lib.linux-x86_64-3.6/torch_scatter/composite
  running build_ext
  building 'torch_scatter._scatter' extension
  creating build/temp.linux-x86_64-3.6
  creating build/temp.linux-x86_64-3.6/tmp
  creating build/temp.linux-x86_64-3.6/tmp/pip-build-rt02l3rf
  creating build/temp.linux-x86_64-3.6/tmp/pip-build-rt02l3rf/torch-scatter
  creating build/temp.linux-x86_64-3.6/tmp/pip-build-rt02l3rf/torch-scatter/csrc
  creating build/temp.linux-x86_64-3.6/tmp/pip-build-rt02l3rf/torch-scatter/csrc/cpu
  gcc -pthread -B /mnt/lustre/share/spring/envs/r0.2.1/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/tmp/pip-build-rt02l3rf/torch-scatter/csrc -I/mnt/lustre/zhangjinghuai1/.local/lib/python3.6/site-packages/torch/include -I/mnt/lustre/zhangjinghuai1/.local/lib/python3.6/site-packages/torch/include/torch/csrc/api/include -I/mnt/lustre/zhangjinghuai1/.local/lib/python3.6/site-packages/torch/include/TH -I/mnt/lustre/zhangjinghuai1/.local/lib/python3.6/site-packages/torch/include/THC -I/mnt/lustre/share/spring/envs/r0.2.1/include/python3.6m -c /tmp/pip-build-rt02l3rf/torch-scatter/csrc/scatter.cpp -o build/temp.linux-x86_64-3.6/tmp/pip-build-rt02l3rf/torch-scatter/csrc/scatter.o -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_scatter -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++11
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  /tmp/pip-build-rt02l3rf/torch-scatter/csrc/scatter.cpp: In static member function ‘static torch::autograd::variable_list ScatterMean::forward(torch::autograd::AutogradContext*, torch::autograd::Variable, torch::autograd::Variable, int64_t, c10::optional<at::Tensor>, c10::optional<long int>)’:
  /tmp/pip-build-rt02l3rf/torch-scatter/csrc/scatter.cpp:98:10: error: ‘class at::Tensor’ has no member named ‘true_divide_’
        out.true_divide_(count);
            ^
  /tmp/pip-build-rt02l3rf/torch-scatter/csrc/scatter.cpp:100:6: error: ‘class at::Tensor’ has no member named ‘floor_divide_’
    out.floor_divide_(count);
        ^
  /tmp/pip-build-rt02l3rf/torch-scatter/csrc/scatter.cpp: In static member function ‘static torch::autograd::variable_list ScatterMean::backward(torch::autograd::AutogradContext*, torch::autograd::variable_list)’:
  /tmp/pip-build-rt02l3rf/torch-scatter/csrc/scatter.cpp:117:13: error: ‘class at::Tensor’ has no member named ‘true_divide_’
       grad_in.true_divide_(count);
               ^
  error: command 'gcc' failed with exit status 1

**Comment by rusty1s** [comment]
Which PyTorch version are you using? The error indicates that you are using 1.4.0, while the code expects >= 1.5.0.

**Comment by Michael1015198808** [comment]
> Which PyTorch version are you using? The error indicates that you are using 1.4.0, while the code expects >= 1.5.0.

Thanks! I also encountered this problem because my Anaconda installed PyTorch v1.3.1 for me, and after manually upgrade PyTorch into v1.7.1, these packages compile gracefully!
